uk.co.compucorp.civicrm.tasksassignments/js/test/mocks/data/task.js

```diff
@@ -0,0 +1,184 @@
+define(function() {
+  return {
+    task: {
```

Comment:
I think there is a big misunderstanding as to what this file is supposed to hold. The mocked data should be the representation of the data *coming from the backend API*, that's all. You should simply have a static, mocked data that you would get in reality from the API.

In here instead i see that you have an example task object, some url encoded data, an array with some tasks objects, etc. They obviously have nothing to do with what you would get *from* the API.

If you want to have a quick way to generate tasks on the fly for testing purposes, then you could think of having a `test/fabricators` folder where you could have `task.js`, `assignment.js`.

Just for not getting the scope of this PR too big, for now `test/fabricators/task.js` could simply return those static object & array you have currently in this file
```js
// test/fabricators/task.js

define([], function () {
  return {

    /**
     * STATIC FOR NOW!
     * 
     * [task description]
     * @return {[type]} [description]
     */
    single: function () {
        return {
          activity_date_time: "2017-04-21",
          assignee_contact_id: ["6"],
          source_contact_id: "205",
          target_contact_id: ["10"],
          details: "<p>Detailed description of the task</p>",
          case_id: null,
          activity_type_id: "88",
          subject: "Sample Task Subject"
        };
    },

    /**
     * STATIC FOR NOW!
     *
     * [tasksList description]
     * @return {[type]} [description]
     */
    list: function () {
      return /* the list you currently have in mock/data/task */
    }
  }
};
```

then in the actual test file you could simply require it and then use it for whatever you need
```js
define([
  test/mocks/fabricators/task
], function (taskFabricator) {
  var fakeTask = taskFabricator.single();
  var fakeTaskList  = taskFabricator.list();
});
```